updating helm to support windows

brycahta opened the pull request:
*Issue #, if available:* N/A

*Description of changes:*
* updates to helm daemonsets, values, and helper to support Windows

*Testing:*
* setup eks cluster with linux and windows nodes
* built and uploaded aemm Windows image to a "test" dockerhub repo
* used my local workspace as the aemm download path for NTH e2e tests
* AEMM pod runs on Windows node successfully
* all NTH e2e tests pass on Windows node using AEMM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

cjerad commented on helm/amazon-ec2-metadata-mock/templates/_helpers.tpl:
```diff
@@ -54,3 +62,41 @@ Create the name of the service account to use
+See https://v1-14.docs.kubernetes.io/docs/setup/release/notes/#deprecations
+*/}}
+{{- define "amazon-ec2-metadata-mock.defaultNodeSelectorTermsPrefix" -}}
+    {{- semverCompare "<1.14" .Capabilities.KubeVersion.Version | ternary "beta.kubernetes.io" "kubernetes.io" -}}
```
1. It turns out that `.Version` is only present in Helm 3.  Here is a [workaround](https://github.com/aws/aws-node-termination-handler/commit/e16065c826fc2fc350e15476ffca34f9222431e8#diff-d4534d4837f947132873af8b9dc3c6a9L74) used to support Helm 2.
2. An issue was encountered when the version includes [build metadata](https://semver.org/#spec-item-10).  Here is a [workaround](https://github.com/aws/aws-node-termination-handler/commit/43a69e0f41f2d7ed97e7199f5026dfc2c7eec6e4#diff-d4534d4837f947132873af8b9dc3c6a9R74).

bwagner5 replied:
Also, might want to do <1.19 since beta is supported on all GA released k8s versions. We recently found out that this is an immutable field in k8s and will not cleanly perform a helm upgrade if it changes.  It's not a big deal here since the use-case is not to constantly upgrade.

cjerad commented on helm/amazon-ec2-metadata-mock/templates/daemonset.windows.yaml:
```diff
@@ -0,0 +1,122 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "amazon-ec2-metadata-mock.fullname" . }}
```
Should this be `amazon-ec2-metadata-mock.fullname.windows`?

brycahta commented:
**Updates from latest push:**
* addressed PR comments
* updating `upload-resources-to-github` to check for existence before adding to list of artifacts to upload
  * needed for Windows
* added Windows binary + image build to travis
* added Windows deployments to travis

**Testing:**
* Submitted changes to a local branch
* Kicked off Windows deployments via Travis on my fork: https://travis-ci.org/github/brycahta/amazon-ec2-metadata-mock/builds/710921663
* Fork'd repo release assets updated: https://github.com/brycahta/amazon-ec2-metadata-mock/releases
* Dockerhub repo images added: https://hub.docker.com/repository/docker/brycahta/dockerhub-testing/tags?page=1

cjerad commented on scripts/push-docker-images:
```diff
@@ -76,7 +76,15 @@ if [[ $MANIFEST == "true" ]]; then
+    manifest_exists=$(docker manifest inspect $IMAGE_REPO:$VERSION > /dev/null ; echo $?)
+    # manifest already exists; therefore, amend is required
+    if [[ manifest_exists -eq 0 ]]; then
+      manifest_create_args+=(--amend)
```
Docker CLI has an open issue ([#954](https://github.com/docker/cli/issues/954)) for the `--amend` flag.  Is this script impacted?

brycahta replied:
saw that as well, but I still saw expected behavior when I tested this locally using my own dockerhub repo. I also do `docker manifest create --amend` first, then proceed with `docker manifest annotate` and finally `docker manifest push` which is a little different from what that guy was doing so possibly not the same issue.
